api/src/integration/java/org/pmiops/workbench/BaseIntegrationTest.java

```diff
@@ -0,0 +1,49 @@
+package org.pmiops.workbench;
+
+import com.google.common.io.Resources;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import org.junit.runner.RunWith;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Primary;
+import org.springframework.context.annotation.Scope;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+
+@RunWith(SpringRunner.class)
+@WebAppConfiguration
+@Import({IntegrationTestConfig.class})
+public abstract class BaseIntegrationTest {
+
+  static WorkbenchConfig config;
```

Comment:
I'd be a bit worried about state carrying over between test cases. Perhaps a setup/tearDown base method integrated approach would be safer. i.e. `setUp()` calls an optional method `getWorkbenchConfig()` (used when there's a consistent config used for the whole test class), `tearDown()` nulls it out. Still allow direct access to `config` in case a subclass needs to customize the config per individual test cases.

Reply:
So the first question for me is whether a subclass is expected to grab this config object and mutate it, or supply its own config bean and import it in with a qualifier. I feel like the latter approach is more idiomatic in Spring. Maybe have a default builder or something for an IntegrationTestConfig for convenience.